### PR TITLE
AST-2914 - Fix:Incompatibility with PHP 8.1 errors.

### PR DESCRIPTION
**Pull request by yssambare12**
### Description

Resolving PHP8.1 incompatibility issue.

### Screenshots

- https://d.pr/v/V2ikO7

```
[25-Apr-2023 14:31:11 UTC] PHP Deprecated:  Automatic conversion of false to array is deprecated in /Users/vrunda1/Local Sites/ai/app/public/wp-content/themes/astra/inc/core/class-astra-icons.php on line 38
[25-Apr-2023 14:31:11 UTC] PHP Deprecated:  Automatic conversion of false to array is deprecated in /Users/vrunda1/Local Sites/ai/app/public/wp-content/themes/astra/inc/core/class-astra-icons.php on line 38
[25-Apr-2023 14:31:11 UTC] PHP Deprecated:  Automatic conversion of false to array is deprecated in /Users/vrunda1/Local Sites/ai/app/public/wp-content/themes/astra/inc/core/builder/class-astra-builder-helper.php on line 1177
[25-Apr-2023 14:31:11 UTC] PHP Deprecated:  Automatic conversion of false to array is deprecated in /Users/vrunda1/Local Sites/ai/app/public/wp-content/themes/astra/inc/extras.php on line 670
[25-Apr-2023 14:31:11 UTC] PHP Deprecated:  Automatic conversion of false to array is deprecated in /Users/vrunda1/Local Sites/ai/app/public/wp-content/themes/astra/inc/extras.php on line 646
[25-Apr-2023 14:31:11 UTC] PHP Deprecated:  Automatic conversion of false to array is deprecated in /Users/vrunda1/Local Sites/ai/app/public/wp-content/themes/astra/inc/core/common-functions.php on line 526
```

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/extras.php [hunk 1]
```diff
@@ -642,8 +642,10 @@ function astra_load_preload_local_fonts( $url, $format = 'woff2' ) {
  * @since 3.6.0
  */
 function astra_get_transparent_header_default_value() {
-	$astra_settings                                      = get_option( ASTRA_THEME_SETTINGS );
-	$astra_settings['transparent-header-default-border'] = isset( $astra_settings['transparent-header-default-border'] ) ? false : true;
+	$astra_settings = get_option( ASTRA_THEME_SETTINGS, array() );
+
+	$astra_settings['transparent-header-default-border'] = isset( $astra_settings['transparent-header-default-border'] ) ? $astra_settings['transparent-header-default-border'] : array();
```
Comment (imnavanath) [inline inc/extras.php:647]: Fallback should be boolean why array is assigned?

File: inc/extras.php [hunk 2]
```diff
@@ -660,15 +662,18 @@ function astra_has_gcp_typo_preset_compatibility() {
...
  * @return string
  */
 function astra_button_default_padding_updated() {
-	$astra_settings                                = get_option( ASTRA_THEME_SETTINGS );
-	$astra_settings['btn-default-padding-updated'] = isset( $astra_settings['btn-default-padding-updated'] ) ? $astra_settings['btn-default-padding-updated'] : true;
-	return apply_filters( 'astra_update_button_padding_defaults', $astra_settings['btn-default-padding-updated'] );
+	$astra_settings  = get_option( ASTRA_THEME_SETTINGS );
+	$padding_updated = isset( $astra_settings['btn-default-padding-updated'] ) ? $astra_settings['btn-default-padding-updated'] : null;
+	if ( $padding_updated === false ) {
+		$padding_updated = array( true );
```
Comment (imnavanath) [inline inc/extras.php:674]: Fallback should be boolean why array is assigned?

File: inc/extras.php [hunk 3]
```diff
@@ -660,15 +662,18 @@ function astra_has_gcp_typo_preset_compatibility() {
...
+	}
```
Comment (imnavanath) [inline inc/extras.php:675]:
```suggestion
	$astra_settings  = get_option( ASTRA_THEME_SETTINGS, array() );
	$padding_updated = isset( $astra_settings['btn-default-padding-updated'] ) ? $astra_settings['btn-default-padding-updated'] : true;
```

Comment (imnavanath) [reply inc/extras.php:675]: Will this work?

File: inc/extras.php (hunk 2 above)
Comment (yssambare12) [reply inc/extras.php]: I was thinking of some other approach but right here boolean fallback is more appropriate @imnavanath ,
Updated accordingly.

File: inc/extras.php (hunk 3 above)
Comment (yssambare12) [reply inc/extras.php]: Yes @imnavanath  your suggestion is working here, this is more flexible than adding if condition.
Thanks for pointing out. Updated accordingly.

File: inc/extras.php (hunk 1 above)
Comment (yssambare12) [reply inc/extras.php]: I was thinking of some other approach but right here boolean fallback is more appropriate @imnavanath ,
Updated accordingly.

File: inc/extras.php [hunk 4]
```diff
@@ -642,8 +642,9 @@ function astra_load_preload_local_fonts( $url, $format = 'woff2' ) {
  * @since 3.6.0
  */
 function astra_get_transparent_header_default_value() {
-	$astra_settings                                      = get_option( ASTRA_THEME_SETTINGS );
-	$astra_settings['transparent-header-default-border'] = isset( $astra_settings['transparent-header-default-border'] ) ? false : true;
+	$astra_settings = get_option( ASTRA_THEME_SETTINGS, array() );
+	$astra_settings['transparent-header-default-border'] = isset( $astra_settings['transparent-header-default-border'] ) ? $astra_settings['transparent-header-default-border'] : false;
```
Comment (imnavanath) [inline inc/extras.php:646]: fallback should be true right???

Comment (yssambare12) [reply inc/extras.php:646]: Oh... Right! It should be true for default behavior.

Comment (yssambare12) [reply inc/extras.php]: Updated.

File: changelog.txt [hunk 5]
```diff
@@ -5,6 +5,7 @@ v4.1.6 (Unreleased)
 - Fix: Header Builder - Menu - Item Divider doesn't work on the last submenu item.
 - Fix: Author Link not working on layout 2 for single post title.
 - Fix: Fatal error: Uncaught TypeError: Unsupported operand types: On frontend and customizer.
+- Fix: Incompatibility depricated warning with PHP v8.1.
```
Comment (imnavanath) [inline changelog.txt:8]:
```suggestion
- Fix: Incompatibility deprecated warning with PHP v8.1.
```

File: inc/extras.php [hunk 6]
```diff
@@ -660,15 +661,15 @@ function astra_has_gcp_typo_preset_compatibility() {
...
  *
  * @since 3.6.3
  * @return string
  */
 function astra_button_default_padding_updated() {
-	$astra_settings                                = get_option( ASTRA_THEME_SETTINGS );
-	$astra_settings['btn-default-padding-updated'] = isset( $astra_settings['btn-default-padding-updated'] ) ? $astra_settings['btn-default-padding-updated'] : true;
-	return apply_filters( 'astra_update_button_padding_defaults', $astra_settings['btn-default-padding-updated'] );
+	$astra_settings  = get_option( ASTRA_THEME_SETTINGS, array() );
+		$padding_updated = isset( $astra_settings['btn-default-padding-updated'] ) ? $astra_settings['btn-default-padding-updated'] : true;
```
Comment (imnavanath) [inline inc/extras.php:671]:
```suggestion
	$astra_settings  = get_option( ASTRA_THEME_SETTINGS, array() );
	$padding_updated = isset( $astra_settings['btn-default-padding-updated'] ) ? $astra_settings['btn-default-padding-updated'] : true;
```

File: inc/core/class-astra-icons.php [hunk 7]
```diff
@@ -34,9 +34,10 @@ public function __construct() {
 	 * @return boolean should be svg or font.
 	 */
 	public static function is_svg_icons() {
-		$astra_settings                               = get_option( ASTRA_THEME_SETTINGS );
-		$astra_settings['can-update-astra-icons-svg'] = ( isset( $astra_settings['can-update-astra-icons-svg'] ) && false === $astra_settings['can-update-astra-icons-svg'] ) ? false : true;
-		return apply_filters( 'astra_is_svg_icons', $astra_settings['can-update-astra-icons-svg'] );
+		$astra_settings = get_option( ASTRA_THEME_SETTINGS );
```
Comment (imnavanath) [inline inc/core/class-astra-icons.php:37]:
```suggestion
		$astra_settings = get_option( ASTRA_THEME_SETTINGS, array() );
```

**Review by imnavanath — APPROVED** [review]

**Review by imnavanath — APPROVED** [review]

**Pull request merged**